### ISSUE TITLE
Error trying to create native image with Graal-rc7

### ISSUE THREAD

**Issue by ilopmar** [micronaut-projects/micronaut-core#682]
This works with Graal-rc6 and Micronaut RC1 but fails with Graal-rc7 and Micronaut RC1 and latest snapshot.

- Clone this repo: https://github.com/micronaut-guides/micronaut-creating-first-graal-app
- Make sure you are using Graal-rc7
- `cd complete`
- `./build-native-image.sh`
- The process fails with:
```
[complete:23540]    classlist:   5,314.81 ms
[complete:23540]        (cap):     938.41 ms
Warning: Detected unnecessary RecomputeFieldValue.ArrayBaseOffset com.oracle.svm.core.jdk.Target_java_nio_DirectByteBuffer.arrayBaseOffset substitution field for java.nio.DirectByteBuffer.arrayBaseOffset. The annotated field can be removed. This ArrayBaseOffset computation can be detected automatically. Use option -H:+UnsafeAutomaticSubstitutionsLogLevel=2 to print all automatically detected substitutions. 
[complete:23540]        setup:   1,939.71 ms
Warning: class initialization of class io.netty.handler.ssl.util.BouncyCastleSelfSignedCertGenerator failed with exception java.lang.NoClassDefFoundError: org/bouncycastle/jce/provider/BouncyCastleProvider. This class will be initialized at runtime because option --report-unsupported-elements-at-runtime is used for image building. Use the option --delay-class-initialization-to-runtime=io.netty.handler.ssl.util.BouncyCastleSelfSignedCertGenerator to explicitly request delayed initialization of this class.
Warning: class initialization of class io.netty.handler.ssl.JdkNpnApplicationProtocolNegotiator failed with exception java.lang.ExceptionInInitializerError. This class will be initialized at runtime because option --report-unsupported-elements-at-runtime is used for image building. Use the option --delay-class-initialization-to-runtime=io.netty.handler.ssl.JdkNpnApplicationProtocolNegotiator to explicitly request delayed initialization of this class.
Warning: class initialization of class io.netty.handler.ssl.ReferenceCountedOpenSslEngine failed with exception java.lang.NoClassDefFoundError: io/netty/internal/tcnative/SSL. This class will be initialized at runtime because option --report-unsupported-elements-at-runtime is used for image building. Use the option --delay-class-initialization-to-runtime=io.netty.handler.ssl.ReferenceCountedOpenSslEngine to explicitly request delayed initialization of this class.
[complete:23540]     analysis:  37,110.85 ms
error: No instances are allowed in the image heap for a class that is initialized or reinitialzied at image runtime: sun.security.provider.NativePRNG
Detailed message:
Error: No instances are allowed in the image heap for a class that is initialized or reinitialzied at image runtime: sun.security.provider.NativePRNG
Trace: 	object io.netty.handler.ssl.util.ThreadLocalInsecureRandom
	method io.netty.handler.ssl.util.ThreadLocalInsecureRandom.current()
Call path from entry point to io.netty.handler.ssl.util.ThreadLocalInsecureRandom.current(): 
	at io.netty.handler.ssl.util.ThreadLocalInsecureRandom.current(ThreadLocalInsecureRandom.java:35)
	at io.netty.handler.ssl.util.SelfSignedCertificate.<init>(SelfSignedCertificate.java:110)
	at io.netty.handler.ssl.util.SelfSignedCertificate.<init>(SelfSignedCertificate.java:88)
	at io.netty.handler.ssl.util.SelfSignedCertificate.<init>(SelfSignedCertificate.java:79)
	at io.micronaut.http.server.netty.ssl.NettyServerSslBuilder.build(NettyServerSslBuilder.java:67)
	at io.micronaut.http.server.netty.NettyHttpServer.<init>(NettyHttpServer.java:207)
	at com.oracle.svm.reflect.proxies.Proxy_io_micronaut_http_server_netty_NettyHttpServer_constructor_io_micronaut_http_server_netty_configuration_NettyHttpServerConfiguration_io_micronaut_context_ApplicationContext_io_micronaut_web_router_Router_io_micronaut_http_server_binding_RequestArgumentSatisfier_io_micronaut_http_codec_MediaTypeCodecRegistry_io_micronaut_http_server_netty_types_NettyCustomizableResponseTypeHandlerRegistry_io_micronaut_web_router_resource_StaticResourceResolver_java_util_concurrent_ExecutorService_java_util_concurrent_ThreadFactory_io_micronaut_scheduling_executor_ExecutorSelector_io_micronaut_http_server_netty_ssl_NettyServerSslBuilder_java_util_List_344.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sun.security.x509.OtherName.getGNI(OtherName.java:132)
	at sun.security.x509.OtherName.equals(OtherName.java:184)
	at java.util.AbstractCollection.contains(AbstractCollection.java:106)
	at java.util.AbstractCollection.containsAll(AbstractCollection.java:318)
	at java.util.JumboEnumSet.containsAll(JumboEnumSet.java:251)
	at com.oracle.svm.core.amd64.AMD64CPUFeatureAccess.verifyHostSupportsArchitecture(AMD64CPUFeatureAccess.java:158)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:138)
	at com.oracle.svm.core.code.CEntryPointCallStubs.com_002eoracle_002esvm_002ecore_002eJavaMainWrapper_002erun_0028int_002corg_002egraalvm_002enativeimage_002ec_002etype_002eCCharPointerPointer_0029(generated:0)

Error: Image building with exit status 1
```

**Comment by graemerocher** [comment]
See https://github.com/oracle/graal/issues/713

**Comment by graemerocher** [comment]
Fixed by 5dd0dc55e97bf534ec8bb8f6a31163558c3564be
